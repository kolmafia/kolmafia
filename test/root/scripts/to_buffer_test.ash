// The following should generate a warning
buffer to_buffer(string initial)
{
    print("initial = '" + initial + "'");
    buffer buf;
    buf.append(initial);
    return buf;
}

buffer buf1 = to_buffer("Initial content 1");
print("buf1 content = '" + buf1.to_string() + "'");

buffer buf1a = to_buffer(buf1);
print("buf1a content = '" + buf1a.to_string() + "'");

buffer buf2 = "Initial content 2";
print("buf2 content = '" + buf2.to_string() + "'");

buffer buf3 = $path[Heavy Rains];
print("buf3 content = '" + buf3.to_string() + "'");
