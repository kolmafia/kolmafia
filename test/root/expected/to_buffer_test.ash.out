Function 'to_buffer(string)' overrides a library function. (to_buffer_test.ash, line 2, char 8 to char 33)
initial = 'Initial content 1'
buf1 content = 'Initial content 1'
buf1a content = 'Initial content 1'
initial = 'Initial content 2'
buf2 content = 'Initial content 2'
initial = 'Heavy Rains'
buf3 content = 'Heavy Rains'
